File: fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum (n: int): int =
    let sumToN = (n*(n+1))/2
    sumToN*sumToN

let sumOfSquares (n: int): int = 
    (n*(n+1)*(2*n+1))/6

let differenceOfSquares (n: int): int = 
    squareOfSum n - sumOfSquares n

File: csharp/difference-of-squares/DifferenceOfSquares.cs
using System;

public static class DifferenceOfSquares
{
    public static int CalculateSquareOfSum(int max)
    {
        int sumToN = (max * (max + 1)) / 2;
        return sumToN * sumToN;
    }

    public static int CalculateSumOfSquares(int max)
    {
        return (max * (max + 1) * (2 * max + 1)) / 6;
    }

    public static int CalculateDifferenceOfSquares(int max)
    {
        return CalculateSquareOfSum(max) - CalculateSumOfSquares(max);
    }
}